specs/language/declarations.tex

```diff
@@ -0,0 +1,44 @@
+\Ch{Declarations}{Decl}
+\Sec{Function Definitions}{Decl.Function}
+\Sec{Attributes}{Decl.Attr}
+\Sub{Entry Attributes}{Decl.Attr.Entry}
+\Sec{Linkage}{Decl.Linkage}
+
+\p An entity that denotes an object, reference, function, type, template, namespace, or value, may have a \textit{linkage}. If a name has \textit{linkage}, it refers to the same entity as the same name introduced by a declaration in another scope. If a variable, function, or another entity with the same name is declared in several scopes, but does not have sufficient \textit{linkage}, then several instances of the entity are generated.
+
+\p There are three linkages recognized: \textit{external linkage}, \textit{internal linkage} and \textit{no linkage}.
+
+\Sub{External Linkage}{Decl.Linkage.External}
+
+\p Entities with \textit{external linkage} can be referred to from the scopes in the other translation units and enable linking between them.
+
+\p The following entities in HLSL have \textit{external linkage}:
+\begin{itemize}
+    \item entry point functions
+    \item functions marked with \texttt{export} keyword
+    \item global variables that are not marked \texttt{static} or \texttt{groupshared} \footnote{These are not really linked with other translation units but rather their values are loaded indirectly based on cbuffer mapping.}
+    \item enumerations
+    \item classes or template classes, their member functions, static data members, nested classes and enumerations
```

Comment:
Do enumerations and classes have external linkage in HLSL?

I think they likely have _internal linkage_ and we duplicate their definitions and use the One Definition Rule to resolve them to a single instance.

I think a static data member of a class is the only thing here that would truly have _external linkage_.